### ISSUE TITLE
Menus broken with Focus follows mouse option.

### ISSUE THREAD

**Issue by ghost** [ValveSoftware/steam-for-linux#6]
Platform: Slackware64 14.0, Xorg 7.7, Xfce 4.10

Problem Description: With "Focus follows Mouse" option all menus and dropdown lists close automatically, when you move the mouse pointer out of it.

How to reproduce: Open "Settings" / "Window Manager" by executing /usr/bin/xfwm4-settings, Select "Focus Tab" and then "Focus follows mouse".

Possible cause: Steam seems to use the wrong X11 window type (top level window instead of pop-up menu window) to draw its menus, so the main Steam window loses focus when opening a menu. Additionally Steam closes the menu window as soon as the menu window loses its focus.

Other commercial applications like Opera and Google Chrome have no issues with their menus.

Why this needs to be fixed: Focus follows mouse is an accessibility option for impaired people.

Additional notes: Maybe this can be reproduced using the Windows version of Steam by going to Control Panel / Easy of Access Center / Make the mouse easier to use / Activate a window by hovering over it with the mouse.

**Comment by rdeforest** [comment]
I've been having this problem for years and just got used to using arrow keys to select from menus. However, I'm now seeing behavior like #3489 when I right-click (e.g. to uninstall a game). I haven't found a work-around yet.

What I don't understand is why my window manager thinks the menu is a new window? Shouldn't it just be a widget like a text field or button? Also, doesn't Steam use WebKit to drive the UI? Why didn't I see this problem in Chrome before they forked it as Blink?

I'm on Ubuntu 14.10, Gnome 3, Steam built 2015-04-13 15:20:04.

**Comment by TwoD** [comment]
Because to X11, everything is a window: http://www.phoronix.com/scan.php?page=article&item=x_wayland_situation

**Comment by nazar-pc** [comment]
@rdeforest, I'm on Steam beta channel and Ubuntu 15.04 (with Unity currently), for quite some time I do not experience this issue at all. Seems to be fixed in Ubuntu or Steam itself, try to update to 15.04 (I live on nightly builds, so, probably it was fixed somewhere during 15.04 development process).

**Comment by rdeforest** [comment]
Nice! I'll switch to the beta channel when I get home. Thanks for the info!

**Comment by TwoD** [comment]
Yes, the beta has improved significantly in this area, though I'm not sure exactly which change[s] did it.

A few other issues seem to have [almost] gone away too.
I used to have major problems with menu items getting stuck half-faded in if I moved out of them too quickly. They then covered everything and prevented screen updates in the covered area. Workaround: find the menu item which was opened, mouse over it again until the menu fades in completely, and then move the mouse and it'll go away properly.

Most links in text content (articles, reviews, etc) used to be unclickable, they now work as they should.

**Comment by rdeforest** [comment]
Confirmed that Ubuntu 15.04 and Steam Beta channel play well together. The
upgrade to 15.04 seems to have fixed an Xorg focus bug:
https://bugs.freedesktop.org/show_bug.cgi?id=26213. I suspect they are
related and that whatever the Xorg people did helped (Steam Beta was still
broken for me on 14.10).

On Mon, Apr 27, 2015 at 10:41 AM, Henrik Danielsson <
notifications@github.com> wrote:

> Yes, the beta has improved significantly in this area, though I'm not sure
> exactly which change[s] did it.
>
> A few other issues seem to have [almost] gone away too.
> I used to have major problems with menu items getting stuck half-faded in
> if I moved out of them too quickly. They then covered everything and
> prevented screen updates in the covered area. Workaround: find the menu
> item which was opened, mouse over it again until the menu fades in
> completely, and then move the mouse and it'll go away properly.
>
> Most links in text content (articles, reviews, etc) used to be
> unclickable, they now work as they should.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ValveSoftware/steam-for-linux/issues/6#issuecomment-96753294>
> .
>



-- 
"I like dark beer, dry cider, peaty scotch, añejo tequila, floral gin,
expensive vodka, and cheap rum." -- A. P.

**Comment by TwoD** [comment]
I still sometimes have this issue, reproduced it today with latest Steam Beta update, in Enlightenment 19.9 on Arch with Xorg at version 1.17.1. It seems to be much more difficult to trigger it though and it only happened a few times and then stopped (same instance of Steam still running).

The way to reproduce it was to Alt+Tab away from Steam just after hoovering something which shows a menu or other popup. The popup starts to fade in but gets stuck on top of everything else until I hoover the same UI item again.

**Comment by inversephase** [comment]
Arch Linux, xfwm 4.12, focus follows mouse, xorg-server 1.17.2-4. Steam client from 7 Oct 2015.

- Moving mouse onto a dropdown menu (for example, STORE) draws the menu. 
- Moving mouse *anywhere* causes the menu to vanish. To clarify:
  - Move mouse 1+ pixels in any direction on the word STORE, menu disappears.
  - Move mouse onto menu below STORE, it disappears before you can use it.

**However** (workaround?) if you move the mouse onto STORE, and to any other pulldown (LIBRARY, COMMUNITY, or even onto one of those then back onto STORE), and then you can use those menus without issue.

Behavior is not random; this can be triggered every time.

**Comment by CarsondaCoolest** [comment]
This issue is almost eight years old and yet I'm still experiencing it.

I am using Linux Peppermint x64 (latest) and the issue is with the dropdown menus that appear when moused over. I have window focus setting set such that simply mousing over the window won't take focus, but any other mouse action (including scrolling) will.

The behaviour is as follows:

1) I hover over the menu item (or click on the menu items at the very top)
2) The submenu appears, !! and takes the focus from the main steam app as if the menu was an app itself. !!
3) I scroll while my mouse is still over the main menu and not the submenu, and the submenu disappears while the focus is given back to the main menu.

I can confirm that similar menus in other applications do not have this issue, and I am inclined to agree with one of the original comments which suggests that "Steam seems to use the wrong X11 window type (top level window instead of pop-up menu window) to draw its menus, so the main Steam window loses focus when opening a menu. Additionally Steam closes the menu window as soon as the menu window loses its focus."

**Comment by viggy96** [comment]
This has been an issue for 9 years now, and it still hasn't been fixed. Come on Valve. With the Steam Deck coming out, there has to be a renewed effort to clean up these bugs on the Steam Linux client.

**Comment by viggy96** [comment]
This issue still occurs on the latest Steam beta, built 23 Sep 2021. I'm running Manjaro GNOME, and I use "click to focus". Even so, when I open a menu, the Steam window loses focus.

**Comment by manen** [comment]
oh yeah. i've been experiencing this so long i thought it was a planned feature

**Comment by ryan-sg** [comment]
Steam Version:  1690583737
Steam Client Build Date:  Fri, Jul 28 11:44 AM UTC -08:00
Steam Web Build Date:  Fri, Jul 28 3:21 PM UTC -08:00
Steam API Version:  SteamClient020

Still happens in 2023.  Gnome 43.4 on Debian/Bookworm.  Workaround is to just turn on click-to-focus temporarily if I need to access right-click, but it sure would be nice to just be able to use right-click menus in steam with sloppy focus.

**Comment by i92guboj** [comment]
Same here also with sloppy focus on Mint/Cinnamon.

Steam Version:  1696019606
Steam Client Build Date:  Fri, Sep 29 7:38 PM UTC -08:00
Steam Web Build Date:  Fri, Sep 29 10:25 PM UTC -08:00
Steam API Version:  SteamClient020

**Comment by Athanasius** [comment]
I'm shocked that this report is almost 12 years old and apparently there's been no move to address it.

Debian bookworm, using fvwm3 with focus follows mouse.  I can't get at the contents of any of the steam client menus.

I'll see if I can add some extra fvwm3 configuration to change focus type for just Steam windows, I guess.

**Comment by Athanasius** [comment]
Actually this appears to have nothing to do with fvwm3 focus configuration.  I'm now using:
```
Style * ClickToFocus
```
temporarily, and the Steam client menus **still close the moment the mouse is moved off the menu label text**.  This happens even with moving the mouse to the right of the `Help` menu label, i.e. it's not because the mouse is being moved down towards the `Store` `Library` `Community` `<my profile name>` 'links' below the menu line.

Because of the issue I can't even tell you the exact Steam client version I'm using, as I can't get to the `Help` > `About Steam` item!  I installed steam today, from scratch, using the Debian 12/bookworm `steam-installer` provided `steam`.  It then downloaded the actual client, so presumably it's the latest.

To be 100% explicit, this appears to only be affecting that top line of menus: `Steam` `View` `Friends` `Games` `Help`.  The `Store` `Library` `Community` `<my profile name>` ones all work fine.  The right click menu for a game in the Library listing also work.  The menu opened from my profile name top-right (next to notifications) also works without issues.

**Comment by Athanasius** [comment]
Aha!  Big Picture mode almost makes this usable as a workaround.

Now I could actually get at things like the system report and diagnostics I can confirm:

```json
  "runtime" : {
    "version" : "0.20240610.91380",
```

**Comment by Athanasius** [comment]
1. I've now set up some key binds in fvwm3 to jump the mouse cursor by 24 pixels, which looks to be enough to get from `Help` into the menu it pops up when clicked.  But on doing that I still end up with the steam client acting on "moused over `Community`" and it opens *that* menu.

2. Looking at this more precisely, the moment the mouse/focus leaves `Menu` it closes its menu, I don't even need to have moved the mouse far enough down to be over `Community` (or another entry on that line).  It's simply that the moment focus is no longer on `Help` its popup menu gets closed.  I've made and uploaded a video to demonstrate this: https://www.youtube.com/watch?v=shLMtp7x7Hc

Basically these top-line menus need to:

1. Have their popup open closer to them in vertical terms.
2. Not have a gap in the focus-following between the actual menu name and the popup.  Focus should move smoothly between the two.

**Comment by IamKayrox** [comment]
<img src="https://media4.giphy.com/media/v1.Y2lkPTc5MGI3NjExYWdjMm4zM3N4b3VheGRqa2g2aWJjb3dvb2NoZTkzZGt5cHY0cWJzciZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/7xZAu81T70Uuc/giphy.gif"></img>
